src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -1325,7 +1328,8 @@ def __init__(self, config: OwlViTConfig):
 
         self.owlvit = OwlViTModel(config)
         self.class_head = OwlViTClassPredictionHead(config)
-        self.box_head = OwlViTBoxPredictionHead(config)
+        self.box_head = OwlViTBoxPredictionHead(config, out_dim=4)
+        self.objectness_head = OwlViTBoxPredictionHead(config, out_dim=1) if config.add_objectness_head else None
```

Comment:
Mmmmm this will change the forward and the architecture we are usually not fond of this. Especially given that you have to break the output to get the correct prediction.  
Might be better to have a new `OwlViTForObjectDetection` no?

Reply:
The change will only happen if `config.add_objectness_head` is `True`. Could you elaborate a bit more about why this is not good?

Reply:
It's typical of what we have been trying to avoid when people add models or want specific usages. The previous model architecture will change based on a new config argument. For starters it's the philosophy of not being a toolbox, thus not accepting small changes like this because we could say "why not add an option for a layer norm for people who use layer norm" and etc etc. If we agree for this one, it's an open door for more changes of this kind and becomes hard to not become bloated. 
Second it's not the same `original` model, so less readable for people who only want the OwlVit. In this case, OwlVit2 is from the same family so adding a new OwlVit2ForObjectDetection is what feels the most natural. 
An nice example is for example UMT5: it's basically T5 but we could change the config and the forward to support relative position embeding in a backward compatible way. But we are polluting the original T5 code with unrelated changes. Event if it's the same family, it's a completely different model.